### ISSUE TITLE
Help with generating contact matrices

### ISSUE THREAD

**Issue by tarak77** [tanlongzhi/dip-c#12]
Thanks again for the contact matrix section! 
1. I believe it should also be easy to format the `contact.pairs` or `impute.pairs` files(from hickit repo) to the required input` .con` files required for `bincon `
2. Again, just to make it more flexible, could we skip the haplotype column in the `.con` files(for haploid cell data) and still generate these contact matrices?

**Comment by tarak77** [comment]
Just a follow up, for the female mouse diploid cell 1CDX1-413, I am trying to generate the contact matrix, but I get the following error:
```
(py2) wg-dhcp174d161d089:CDX1-413.20k tarakshisode$ ../dip-c bincon -b 5000000 -H -l ../mm10.chr.len GSE117109_GSM2476407_1CDX1-413.dedup.con.gz > GSE117109_GSM2476407_1CDX1-413.dedup.bincon.txt
[M::bincon] read 284930 putative contacts (92.16% intra-chromosomal, 51.65% legs phased)
[M::bincon] generating a 547 x 547 matrix
Traceback (most recent call last):
  File "../dip-c", line 122, in <module>
    main()
  File "../dip-c", line 55, in main
    return_value = bincon.bincon(sys.argv[1:])
  File "/Users/tarakshisode/dip-c/bincon.py", line 118, in bincon
    matrix_data = con_data_to_matrix(con_data, hom_offsets, matrix_bin_size, matrix_size, merge_haplotypes, display_num_cons)
  File "/Users/tarakshisode/dip-c/bincon.py", line 32, in con_data_to_matrix
    matrix_index_1, matrix_index_2 = con_to_matrix_index(con, hom_offsets, matrix_bin_size, merge_haplotypes)
  File "/Users/tarakshisode/dip-c/bincon.py", line 19, in con_to_matrix_index
    matrix_index_1 = leg_to_matrix_index(con.leg_1(), hom_offsets, matrix_bin_size, merge_haplotypes)
  File "/Users/tarakshisode/dip-c/bincon.py", line 16, in leg_to_matrix_index
    return hom_offsets[hom_name] + int(round(float(ref_locus) / matrix_bin_size))
KeyError: 'GL456210.1(pat)'
```
I then filter out the `deduce.con` file to remove such chromosomes, then again I get:

```
(py2) wg-dhcp174d161d089:CDX1-413.20k tarakshisode$ ../dip-c bincon -b 5000000 -H -l ../mm10.chr.len GSE117109_GSM2476407_1CDX1-413.dedup.new.con.gz > GSE117109_GSM2476407_1CDX1-413.dedup.bincon.txt
[M::bincon] read 284505 putative contacts (92.26% intra-chromosomal, 51.7% legs phased)
[M::bincon] generating a 547 x 547 matrix
[M::bincon] processed 10000 contacts
[M::bincon] processed 20000 contacts
Traceback (most recent call last):
  File "../dip-c", line 122, in <module>
    main()
  File "../dip-c", line 55, in main
    return_value = bincon.bincon(sys.argv[1:])
  File "/Users/tarakshisode/dip-c/bincon.py", line 118, in bincon
    matrix_data = con_data_to_matrix(con_data, hom_offsets, matrix_bin_size, matrix_size, merge_haplotypes, display_num_cons)
  File "/Users/tarakshisode/dip-c/bincon.py", line 32, in con_data_to_matrix
    matrix_index_1, matrix_index_2 = con_to_matrix_index(con, hom_offsets, matrix_bin_size, merge_haplotypes)
  File "/Users/tarakshisode/dip-c/bincon.py", line 20, in con_to_matrix_index
    matrix_index_2 = leg_to_matrix_index(con.leg_2(), hom_offsets, matrix_bin_size, merge_haplotypes)
  File "/Users/tarakshisode/dip-c/bincon.py", line 16, in leg_to_matrix_index
    return hom_offsets[hom_name] + int(round(float(ref_locus) / matrix_bin_size))
KeyError: 'chrY(pat)'
```

I filter out the `chrY` from `dedup.con` file too, but still it gives the error?

Same happens for `chrM `.

Does `bincon.py` take into account the mouse female cells?

**Comment by tanlongzhi** [comment]
I recently fixed a bug (87c50f93ab9e1dc54b67af6804f47dbf8d9a1a83) so that `dip-c bincon` no longer throws an error when a chromosome is not found in the `chr.len` file. Hope that helps.

**Comment by tarak77** [comment]
Great!
